test/Interop/Cxx/value-witness-table/Inputs/basic.h

```diff
@@ -0,0 +1,29 @@
+#ifndef TEST_INTEROP_CXX_DESTRUCTOR_INPUTS_BASIC_H
+#define TEST_INTEROP_CXX_DESTRUCTOR_INPUTS_BASIC_H
+
+struct HasUserProvidedDestructor {
+  int *value;
+  ~HasUserProvidedDestructor() { *value = 42; }
```

Comment:
I'm not sure, but I think it's legal for the C++ compiler to decide to optimize away the store to `*value`, because it's undefined behavior to try to access that value after the destructor has run.

Instead, I would give the destructor some side effect that is observable from outside the object being destroyed.

Reply:
Please disregard. I completely missed the fact that `value` is a pointer and that the value being modified lives outside the object. This is fine.